File: h2o-algos/src/main/java/water/userapi/DataException.java
package water.userapi;

/**
 * Wrapper for all kinds of H2O and IO exceptions
 * 
 * Created by vpatryshev on 2/20/17.
 */
public class DataException extends RuntimeException {
  final Exception cause;
  final String message;
  
  public DataException(String message, Exception cause) {
    this.message = message;
    this.cause = cause;
  }

  public DataException(Exception cause) {
    this.message = cause.getMessage();
    this.cause = cause;
  }

  public DataException(String message) {
    this.message = message;
    this.cause = null;
  }
  
  
}

File: h2o-scala/src/main/scala/water/userapi/DataException.scala
package water.userapi

/**
  * Wrapper for all kinds of H2O and IO exceptions
  * 
  * Created by vpatryshev on 3/26/17.
  */
case class DataException(val message: String, 
                         val cause: Option[Exception] = None) 
  extends RuntimeException {

  def this(message: String, cause: Exception) = this(message, Some(cause))
  def this(cause: Exception) = this(cause.getMessage, cause)
}

object DataException {
  def apply(message: String, cause: Exception): DataException = 
    new DataException(message, Some(cause))
  
  def apply(cause: Exception): DataException = 
    apply(cause.getMessage, cause)
}

